src/main/java/net/sf/jabref/gui/PreviewPanel.java

```diff
@@ -273,13 +272,19 @@ public void update() {
         if (layout.isPresent()){
             StringBuilder sb = new StringBuilder();
             bibEntry.ifPresent(entry -> sb.append(layout.get()
-                    .doLayout(entry, databaseContext.map(BibDatabaseContext::getDatabase).orElse(null), highlightPattern)));
+                    .doLayout(entry, databaseContext.map(BibDatabaseContext::getDatabase).orElse(null))));
             setPreviewLabel(sb.toString());
+            markHighlights();
         }
         else if (basePanel.isPresent()){
             citationStyleWorker = Optional.of(new CitationStyleWorker(this, previewPane));
             citationStyleWorker.get().execute();
         }
+
+    }
+
+    public void markHighlights() {
```

Comment:
Does it make sense to set the highlight pattern automatically as a last step in `setPreviewLabel` ?